deploy/scripts/environment_check.sh

```diff
@@ -65,45 +144,105 @@ wait_ds_ready() {
     local desiredNumberScheduled=$(echo $ds | jq .status.desiredNumberScheduled)
 
     if [ "$desiredNumberScheduled" == "$numberReady" ] && [ "$desiredNumberScheduled" != "0" ]; then
-      echo "all pods ready ($numberReady/$desiredNumberScheduled)"
+      info "all pods ready ($numberReady/$desiredNumberScheduled)"
       return
     fi
 
-    echo "waiting for pods to become ready ($numberReady/$desiredNumberScheduled)"
+    info "waiting for pods to become ready ($numberReady/$desiredNumberScheduled)"
     sleep 3
   done
 }
 
-validate_ds() {
+check_mount_propagation() {
   local allSupported=true
   local pods=$(kubectl -l app=longhorn-environment-check get po -o json)
 
-  for ((i=0; i<1; i++)); do
+  local ds=$(kubectl get ds/longhorn-environment-check -o json)
+  local desiredNumberScheduled=$(echo $ds | jq .status.desiredNumberScheduled)
+
+  for ((i=0; i<desiredNumberScheduled; i++)); do
     local pod=$(echo $pods | jq .items[$i])
     local nodeName=$(echo $pod | jq -r .spec.nodeName)
     local mountPropagation=$(echo $pod | jq -r '.spec.containers[0].volumeMounts[] | select(.name=="mountpoint") | .mountPropagation')
 
     if [ "$mountPropagation" != "Bidirectional" ]; then
       allSupported=false
-      echo "node $nodeName: MountPropagation DISABLED"
+      error "node $nodeName: MountPropagation DISABLED"
     fi
   done
 
   if [ "$allSupported" != "true" ]; then
-    echo
-    echo "  MountPropagation is disabled on at least one node."
-    echo "  As a result, CSI driver and Base image cannot be supported."
-    echo
+    error "MountPropagation is disabled on at least one node. As a result, CSI driver and Base image cannot be supported."
     exit 1
   else
-    echo -e "\n  MountPropagation is enabled!\n"
+    info "MountPropagation is enabled!"
+  fi
+}
+
+check_package_installed() {
+  local packages=($@)
+  local pods=$(kubectl get pods -o name | grep longhorn-environment-check)
+
+  local allFound=true
+
+  for ((i=0; i<${#packages[@]}; i++)); do
+    local package=${packages[$i]}
+
+    for pod in ${pods}; do
+      kubectl exec -it $pod -- nsenter --mount=/proc/1/ns/mnt -- bash -c "$CHECK_CMD $package" > /dev/null 2>&1
+      if [ $? != 0 ]; then
+        allFound=false
+        node=`kubectl get ${pod} --no-headers -o=custom-columns=:.spec.nodeName`
+        error "Not found: $package in $pod"
+      fi
+    done
+  done
+
+  if [ "$allFound" == "false" ]; then
+    error "Please install missing packages."
+    exit 2
   fi
 }
 
-dependencies kubectl jq mktemp
+OS=`detect_os`
+if [ x"$OS" == x"" ]; then
+  error "Unable to detect OS"
+  exit 2
+fi
+
 TEMP_DIR=$(mktemp -d)
+
+case $OS in
+*"debian"* | *"ubuntu"* )
+  CHECK_CMD="dpkg-query -l"
+  packages=(nfs-common open-iscsi)
+  ;;
+*"centos"* | *"fedora"* | *"rocky"*)
+  CHECK_CMD="rpm -q"
+  packages=(nfs-utils iscsi-initiator-utils)
+  ;;
+*"suse"* )
+  CHECK_CMD="zypper se -i"
+  packages=(nfs-client open-iscsi)
+  ;;
+*)
+  CHECK_CMD=""
+  packages=()
```

Comment:
If not supported OS, suggest to error out and exit.

Reply:
Okay.

Reply:
> > > LGTM, but it seems we don't check if multipass is running, do we need?
> > 
> > 
> > I'm thinking how to check multipath on the node. Maybe just check the `multipath.conf` has a correct blacklist? [longhorn.io/kb/troubleshooting-volume-with-multipath](https://longhorn.io/kb/troubleshooting-volume-with-multipath/)
> > cc. @PhanLe1010
> 
> or just check if the multipass is running, show a warning to inform users there will be probably some problem about volumes automatically mounted? I don't think we need to specially identify the problem. WDYT?

Good idea!